src/pal/src/map/virtual.cpp

```diff
@@ -910,6 +930,66 @@ static LPVOID VIRTUALReserveMemory(
 
     InternalEnterCriticalSection(pthrCurrent, &virtual_critsec);
 
+    // If this is a request for special executable (JIT'ed) memory then, first of all,
+    // try to get memory from the executable memory allocator to satisfy the request.
+    if (((flAllocationType & MEM_RESERVE_EXECUTABLE) != 0) && (lpAddress == NULL))
+    {
+        pRetVal = g_executableMemoryAllocator.AllocateMemory(MemSize);
```

Comment:
The current code has a bug - if somebody asks for 4.1 GB, you will actually only allocate 0.1GB because of the integer gets truncated.

Yes, changing AllocateMemory to take size_t would be definitely cleaner.